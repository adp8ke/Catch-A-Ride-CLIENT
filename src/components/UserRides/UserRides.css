.driverRides, .passengerRides {
  list-style-type: none;
  padding-left: 0px;
}

.divide {
  color: white;
  font-size: 28px;
  width: 60%;
  border-bottom:solid 4px white;
  margin:0 auto;
}

.rideLink {
  text-decoration: none;
  color: white;
}
.errorMessage{
  background:rgba(25,31,56,0.7);
  width:50%;
  margin:0 auto;
  padding-top:15px;
  padding-bottom:15px;
  font-size: 20px;
  margin-top:10px;
  border-radius:5px;
}
.myrides{
  font-size:34px;
}

@media only screen and (min-width: 700px) {
  .driverRides, .passengerRides {
    display: flex;
    flex-flow: row wrap;
    align-content: space-around;
    justify-content: center;
  }

  .divide {
    width: 33%;
  }
}

@media only screen and (min-width: 500px) {
  .driverRides, .passengerRides {
    display: flex;
    flex-flow: row wrap;
    align-content: space-around;
    justify-content: center;
  }
}