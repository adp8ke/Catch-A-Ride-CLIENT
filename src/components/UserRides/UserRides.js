import React from 'react';
import {Link} from 'react-router-dom';
import DriversApiService from '../../services/RidesService/rides-driver-service';
import PassengersApiService from '../../services/RidesService/rides-passenger-service';
import RideContext from '../../context/RideContext';
import Ride from '../Ride/Ride';
import './UserRides.css';

export default class UserRides extends React.Component {
  static contextType = RideContext;

  state = {
    driverError: null,
    passError: null
  };

  componentDidMount() {
    DriversApiService.getDriverRides()
      .then(res => {
        this.context.setDriverRides(res);
      })
      .catch(res => {
        this.context.clearDriverRides();
        this.setState({driverError: res.error});
      });
    PassengersApiService.getAllRides()
      .then(res => {
        this.context.setPassengerRides(res);
      })
      .catch(res => {
        this.context.clearPassengerRides();
        this.setState({passError: res.error});
      });
  }

  driverRidesList() {
    const driverRides = this.context.driverRides.map((ride, i) => (
      <li className='ride' key={i}><Link className='rideLink' to={`/rides/${ride.id}`}><Ride ride={ride} /></Link></li>
    ));
    return (
      <ul className='driverRides'>
        {driverRides}
      </ul>
    );
  }

  passengerRidesList() {
    const passengerRides = this.context.passengerRides.map((ride, i) => (
      <li className='ride' key={i}><Link className='rideLink' to={`/rides/${ride.id}`}><Ride ride={ride} /></Link></li>
    ));
    return (
      <ul className='passengerRides'>
        {passengerRides}
      </ul>
    );
  }

  render() {
    // console.log('context error',this.context.error);
    const { driverError, passError } = this.state;
    return (
      <>
        <h2>My Rides</h2>
        <h3 className='divide'>Driver</h3>
        {driverError && <p className='error'>{driverError}</p>}
        {this.driverRidesList()}
        <h3 className='divide'>Passenger</h3>
        {passError && <p className='error'>{passError}</p>}
        {this.passengerRidesList()}
      </>
    );
  }
}
