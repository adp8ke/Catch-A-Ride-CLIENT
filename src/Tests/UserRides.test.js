import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';
import UserRides from '../components/UserRides/UserRides';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemoryRouter><UserRides /></MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<MemoryRouter><UserRides /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});