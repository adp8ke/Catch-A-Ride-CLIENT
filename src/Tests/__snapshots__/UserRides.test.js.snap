// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <h2
    className="myrides elegantshadow"
  >
    My Rides
  </h2>,
  <h3
    className="divide"
  >
    Driver
  </h3>,
  <ul
    className="driverRides"
  />,
  <h3
    className="divide"
  >
    Passenger
  </h3>,
  <ul
    className="passengerRides"
  />,
]
`;
